connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaOffsetBackingStore.java

```diff
@@ -316,6 +316,9 @@ public Map<ByteBuffer, ByteBuffer> convert(Void result) {
         // if we've already read up to the end. However, it also should not be common (offsets should only be read when
         // resetting a task). Always requiring that we read to the end is simpler than trying to differentiate when it
         // is safe not to (which should only be if we *know* we've maintained ownership since the last write).
+        if (offsetLog.unexpectedExceptionCaught.get()) {
+            throw new ConnectException("Unexpected Exception caught", offsetLog.exception);
+        }
```

Comment:
QQ - what if ConnectException is thrown inside `readToEnd` method instead in next line ? We may not need to access variables from KafkaBasedLog class explicitly.

Reply:
Connect Exception can be thrown inside the readToEnd method, we need to handle it at all place where it is called, instead of just in KafkaOffsetBackingStore. It would be better approach as it would handle all cases of this bug.